core/src/action.rs

```diff
@@ -73,10 +73,13 @@ impl Hash for ActionWrapper {
 pub enum Action {
     /// entry to Commit
     /// MUST already have passed all callback checks
-    Commit(Entry),
+    Commit((Entry, Option<Address>)),
     /// GetEntry by address
     GetEntry(Address),
-
+    ///
+    UpdateEntry((Address, Address)),
```

Comment:
Why the extra `()`

Reply:
Its an enum variant, which is a tuple.
Don't know any other way to make the unwrap_to! macro work.

Reply:
I don't think it's a good idea to make this unnecessary change just because `unwrap_to!` doesn't work with it.

Reply:
nvm 😁 I saw the usage of this and really like it.